packages/sign-client/src/controllers/engine.ts

```diff
@@ -1283,36 +1301,42 @@ export class Engine extends IEngine {
   // ---------- Relay Events Router ----------------------------------- //
 
   private registerRelayerEvents() {
-    this.client.core.relayer.on(
-      RELAYER_EVENTS.message,
-      async (event: RelayerTypes.MessageEvent) => {
-        const { topic, message } = event;
-
-        // Retrieve the public key (if defined) to decrypt possible `auth_request` response
-        const { publicKey } = this.client.auth.authKeys.keys.includes(AUTH_PUBLIC_KEY_NAME)
-          ? this.client.auth.authKeys.get(AUTH_PUBLIC_KEY_NAME)
-          : ({ responseTopic: undefined, publicKey: undefined } as any);
-
-        const payload = await this.client.core.crypto.decode(topic, message, {
-          receiverPublicKey: publicKey,
-        });
+    this.client.core.relayer.on(RELAYER_EVENTS.message, (event: RelayerTypes.MessageEvent) => {
+      // capture any messages that arrive before the client is initialized so we can process them after initialization is complete
+      if (!this.initialized || this.relayMessageCache.length > 0) {
```

Comment:
Is it a cache or a queue ? Why are we pushing the events in the cache instead of processing them?

Reply:
we put all messages received during init, and process them after init has finished. Otherwise it can lead to missed events by the implementing app